### ISSUE TITLE
calculate WER

### ISSUE THREAD

**Issue by ahkarami** [NVIDIA/NeMo#3318]
Hi,
Thanks for your great repo. I have 2 questions:
1- In the Nemo repo is there any implemented code for calculating the WER (word error rate) for a data set & model's transcription? (_i.e._, I mean for example I want to calculate the WER of a trained model on a test set, which I have its ground truth in a text file. So, is there any script in Nemo that I can use it? or I must use other codes such as: [jiwer](https://pypi.org/project/jiwer/))
2- In the [`ASR_CTC_Language_Finetuning`](https://colab.research.google.com/github/NVIDIA/NeMo/blob/stable/tutorials/asr/ASR_CTC_Language_Finetuning.ipynb#scrollTo=iK-RQXEZfq1V) tutorial, if one enables `use_cer` (character error rate) parameter, then the tensor board results will be shown the CER instead of WER? (In other words, I have used the `use_cer = True`, and for example the tensor board showed me the **WER** of the trained model on epoch=31 on validation set is ~ 0.1028. Now I want to know that it is really WER or it is CER? and if it is CER, how I can calculate the WER of the model on my validation or test set?) 
![val_WER](https://user-images.githubusercontent.com/28620609/145693294-daf62515-e6a7-4ff7-a4f8-defdf0eb629b.png)

Best

**Comment by titu1994** [comment]
For (1), you can use https://github.com/NVIDIA/NeMo/blob/main/examples/asr/transcribe_speech_parallel.py for the time being. We will be refactoring the ASR example scripts in the coming month or two to better support newer model types in a generic way. Or you can use jiwer after computing the transcriptions using the transcribe_speech.py and writing a short script. 

For (2), the log name remains the same (*_wer) but in fact it does compute CER and shows the results there. This is obviously confusing, we will improve this to explicitly log CER or WER in the name as well.

If after training, you want to switch batch to wer, you will need to override model._wer.use_cer = False where model is the trained ASR model.

**Comment by ahkarami** [comment]
Dear @titu1994 
Thank you very much. As always, you answered very quickly, clearly and completely. One of the great advantages of your framework (Nemo) is its excellent responsiveness and support. I hope to see more success of this framework in the future.
Best

**Comment by titu1994** [comment]
Once this is merged, https://github.com/NVIDIA/NeMo/pull/3334 should have a nice script to evaluate any ASR model. 
About the CER / WER logging, i took a look at it and its somewhat problmatic. Changing one dynamically means that the exp_manager metric to be logged must also be changed, but the model cant control that. In the worst case, it will cause a long 1 epoch training run and then crash when trying to save checkpoint since it will log metric as `val_wer` but model will emit `val_cer`. 

We will discuss if this breaking change is necessary, or cosmetic (albeit confusing).

**Comment by ahkarami** [comment]
Thanks for your great response. I have 2 notes:
1- I think it would be great if you could add `confidence interval` for evaluation of ASR models (as I heard some other frameworks like Kaldi have it). 
2- for `val_wer` or `val_cer` I have changed the code as below:
```python
    # set the Nemo Experiment Manager: 
    if use_cer:
        monitor_str = "val_cer"
    else:
        monitor_str = "val_wer"
    config = exp_manager.ExpManagerConfig(
    exp_dir=Config.exp_dir,
    name=Config.name_exp_manager,
    checkpoint_callback_params=exp_manager.CallbackParams(
        monitor=monitor_str,
        mode="min",
        always_save_nemo=True,
        save_best_model=True,
    ),
    )
```
this `monitor=monitor_str` assignment doesn't address the problem?
Best

**Comment by titu1994** [comment]
(2) requires config changes + code changes to every single model, which is why it's a large and breaking change. 

We don't have confidence interval support on our list, not all models can support it and it's not a frequent use case

**Comment by owos** [comment]
![image](https://github.com/NVIDIA/NeMo/assets/48519427/4c2f09a7-2159-469d-9564-3993afbe278c)
Please is this 1176 the WER or I should be looking at something else, because the 1176% looks strangely very high to me.
Also, is there a log file I can go look at whihc has a summary of my just concluded training?

**Comment by owos** [comment]
Ohh, I figured as the numbers we're reducing. I'm not using tensorboard or W&B but I figured that every saved checkpoint has the wer added to the file name.

**Comment by titu1994** [comment]
Then you're calculating Word erore rate instead of char error rate

**Comment by owos** [comment]
I understand this. I think my question should have been how does using WER to eval during finetuning affects the finetuned model's performance?
